jOOL-java-8/src/main/java/org/jooq/lambda/tuple/Tuple2.java

```diff
@@ -264,15 +276,76 @@ public final Tuple0 skip2() {
      * </pre></code>
      */
     public static final <T extends Comparable<? super T>> Optional<Tuple2<T, T>> intersect(Tuple2<T, T> left, Tuple2<T, T> right) {
-        if (overlaps(left, right))
+        if (overlaps(left, right)) {
+            if (!left.finite() || !right.finite()) {
+                // todo?
+                throw new UnsupportedOperationException("to be done in another commit!");
+            }
             return Optional.of(new Tuple2<>(
-                left.v1.compareTo(right.v1) >= 0 ? left.v1 : right.v1,
-                left.v2.compareTo(right.v2) <= 0 ? left.v2 : right.v2
+                    left.v1.compareTo(right.v1) >= 0 ? left.v1 : right.v1,
+                    left.v2.compareTo(right.v2) <= 0 ? left.v2 : right.v2
             ));
-        else
+        } else
             return Optional.empty();
     }
 
+    /**
+     * Whether the given values are included in the tuple. Assumes inclusiveness.
+     * <p>
+     * <code><pre>
+     * // true
+     * contains(range(1, 3), 1, 3)
+     * contains(range(2, null), 5)
+     *
+     * // false
+     * contains(range(1, 3), 2, 4)
+     * contains(range(2, null), 1)
+     * </pre></code>
+     */
+    @SafeVarargs
+    public static final <T extends Comparable<? super T>> boolean contains(Tuple2<T, T> tuple2, T... vals) {
+        return Arrays.stream(vals).map(v -> {
+            if (tuple2.v1 != null && tuple2.v1.compareTo(v) > 0)
+                return false; // value is below the lower bound
+            if (tuple2.v2 != null && tuple2.v2.compareTo(v) < 0)
+                return false; // value is above the upper bound
+            return true;
+        }).reduce(Boolean::logicalAnd).orElse(false);
+    }
+
+    /**
+     * Whether the right tuple is included in the left tuple.
+     * <p>
+     * <code><pre>
+     * // true
+     * contains(range(1, 3), range(1, 2))
+     * contains(range(2, null), range(3, 5))
+     *
+     * // false
+     * contains(range(1, 3), range(null, 2))
+     * contains(range(2, null), range(1, 3))
+     * </pre></code>
+     */
+    public static final <T extends Comparable<? super T>> boolean contains(Tuple2<T, T> left, Tuple2<T, T> right) {
+        return contains(left, right.v1, right.v2);
+    }
+
+    public final <T extends Comparable<? super T>> boolean lowerInfinite() {
+        return v1 == null;
+    }
+
+    public final <T extends Comparable<? super T>> boolean upperInfinite() {
+        return v2 == null;
+    }
+
+    public final <T extends Comparable<? super T>> boolean finite() {
+        return !lowerInfinite() && !upperInfinite();
+    }
+
+    public final <T extends Comparable<? super T>> boolean infinite() {
+        return lowerInfinite() && upperInfinite();
+    }
```

Comment:
All of this added API:

- Should be added to `Range` only, not to `Tuple2`. A `Tuple2` has no notion of one of its attributes being "infinite", just because it happens to be `null`.
- In my opinion doesn't deserve to be public API anyway.